cmd/controller-manager/app/controller_manager.go

```diff
@@ -187,10 +190,25 @@ func Run(controllerManagerOptions *options.ControllerManagerServer) error {
 
 	// Create event broadcaster
 	glog.V(4).Info("Creating event broadcaster")
+	eventsScheme := runtime.NewScheme()
+	// We use ConfigMapLock/EndpointsLock which emit events for ConfigMap/Endpoints and hence we need core/v1 types for it
+	if err = corev1.AddToScheme(eventsScheme); err != nil {
+		return err
+	}
+	// We also emit events for our own types
+	if err = servicecatalogv1beta1.AddToScheme(eventsScheme); err != nil {
+		return err
+	}
+	if err = settingsv1alpha1.AddToScheme(eventsScheme); err != nil {
+		return err
+	}
+
 	eventBroadcaster := record.NewBroadcaster()
-	eventBroadcaster.StartLogging(glog.Infof)
-	eventBroadcaster.StartRecordingToSink(&v1core.EventSinkImpl{Interface: k8sKubeClient.CoreV1().Events("")})
-	recorder := eventBroadcaster.NewRecorder(api.Scheme, v1.EventSource{Component: controllerManagerAgentName})
+	loggingWatch := eventBroadcaster.StartLogging(glog.Infof)
+	defer loggingWatch.Stop()
+	recordingWatch := eventBroadcaster.StartRecordingToSink(&v1core.EventSinkImpl{Interface: k8sKubeClient.CoreV1().Events("")})
+	defer recordingWatch.Stop()
+	recorder := eventBroadcaster.NewRecorder(eventsScheme, v1.EventSource{Component: controllerManagerAgentName})
```

Comment:
is there any "wait to be ready" type function that needs to be called? I'm thinking of how informers work. I think because this is a push, where that is a watch, we're fine without any such call being necessary.

Reply:
No, afaik. Events work for instances and bindings, for example.

Reply:
That's my understanding, yes.